### ISSUE TITLE
Getting error "vagrant@127.0.0.1: Permission denied (publickey)." when provisioning or trying to ssh.

### ISSUE THREAD

**Issue by MeanEYE** [hashicorp/vagrant#12693]
### Vagrant version

Vagrant 2.2.19

### Host operating system

Debian Testing

### Guest operating system

Debian Jessie

### Vagrantfile

```ruby
Vagrant.configure('2') do |config|
	config.vm.box = 'debian/contrib-jessie64'

	# customize virtual machine
	config.vm.provider 'virtualbox' do |vm|
		# name our machine for easier management
		path = File.dirname(__FILE__)
		name = File.basename(path)

		if name == 'Site' or name == 'Web'
			vm.name = File.basename(File.dirname(path))
		else
			vm.name = name
		end

		# configure virtualization options
		vm.customize [ 'modifyvm', :id, '--paravirtprovider', 'kvm' ]

		# configure virtual machine resources
		vm.memory = 256
		vm.cpus = 1
	end

	# install web server and required components
	config.vm.provision :ansible do |ansible|
		ansible.compatibility_mode = '2.0'
		ansible.playbook = 'provision/playbook.yml'
		ansible.limit = 'all'
		ansible.groups = {
			'all' => ['caracal']
		}
	end

	# configure synchronized folder to specific user
	config.vm.synced_folder 'site', '/vagrant', owner: 'www-data', group: 'www-data'

	# configure network
	config.vm.network :forwarded_port, host:8080, guest:80
end
```

### Debug output

https://gist.github.com/MeanEYE/01a3107bfb66c4d672d83b23e609745b

### Expected behavior

Provisioning to complete and `vagrant ssh` to log me into guest machine.

### Actual behavior

Message `vagrant@127.0.0.1: Permission denied (publickey).` is printed and nothing happens afterwards.

### Steps to reproduce

1. Execute `vagrant up` or `vagrant ssh`.

### Additional steps tried to solve the issue

- Setting `insert_key` to false and messing with private keys;
- Reinstalled both VirtualBox and Vagrant;
- Removed configurations for Vagrant and VirtualBox;
- Destroyed guest machine multiple times;
- Using packaged tools instead of system tools (ssh and others);
- Checking and resetting permissions on keys and `authorized_keys` inside of guest machine;

In short, after update Vagrant boxes just stopped working and I can't seem to find a way to make them work.

**Comment by MeanEYE** [comment]
Tried switching to `debian/contrib-buster64` and it worked. However others are not. Am not sure if this issue is relevant anymore, so I am closing it. Reopen if there's a need!

**Comment by MeanEYE** [comment]
Temporary fix is to add `PubkeyAcceptedKeyTypes +ssh-rsa` to `~/.ssh/config`.
